CONTRIBUTING.md

```diff
@@ -103,6 +103,12 @@ for review. Use the following simple PR templates to structure the PR:
 
 * [PR Template](https://github.com/asyml/forte/blob/master/.github/PULL_REQUEST_TEMPLATE.md)
 
+Here is a detailed explanation of each section in the template:
+* The first line should always start with `This PR fixes [issue link]` where `[issue link]` can be replaced with a `#ISSUE_ID` associated to a specific [issue](https://github.com/asyml/forte/issues). This allows Github to automatically link your pull request to the corresponding issue.
+* **Description of changes**: You should include a description of the changes that you make. If the pull request is aimed to fix an issue, you can explain the approaches to address the problem.
+* **Possible influences of this PR**: List all the potential side-effects of your update. Examples include influences on compatibility, performance, API signature, etc.
+* **Test Conducted**: Describe the test cases to verify the changes in pull request. You should create unit tests for any newly added features. We will use [Covergage](https://coverage.readthedocs.io/en/6.3/) to gauge the effectiveness of tests.
```

Comment:
Let's make it clear that unit tests are always required and all branches should be tested